src/c++/perf_analyzer/test_infer_context.cc

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "doctest.h"
+#include "infer_context.h"
+#include "mock_data_loader.h"
+#include "mock_infer_data_manager.h"
+#include "mock_sequence_manager.h"
+
+namespace triton { namespace perfanalyzer {
+
+class TestInferContext : public InferContext {
+ public:
+  void UpdateSeqJsonData(size_t seq_stat_index)
+  {
+    InferContext::UpdateSeqJsonData(seq_stat_index);
+  }
+
+  std::shared_ptr<SequenceManager>& sequence_manager_{
+      InferContext::sequence_manager_};
+  std::shared_ptr<DataLoader>& data_loader_{InferContext::data_loader_};
+  std::shared_ptr<IInferDataManager>& infer_data_manager_{
+      InferContext::infer_data_manager_};
+  std::shared_ptr<ThreadStat>& thread_stat_{InferContext::thread_stat_};
+};
+
+TEST_CASE("update_seq_json_data: testing the UpdateSeqJsonData function")
```

Comment:
Can you add a comment about what this test is doing and testing?

Reply:
I think I would feel better if we added a test to test_request_rate_manager or test_concurrency_manager that used the real data and confirmed the real data came out correctly. That way we can fully confirm that we wrap around, regardless of number of threads, etc.

Reply:
Added comment 👍

Updated test to call `SendSequenceInferRequest()` several times and record the json data step id to confirm round robin order 👍